### PR TITLE
Add more docs on assembly

### PR DESCRIPTION
**Pull request by etspaceman**
Hey there - 

@kubukoz and I ran into this problem when running `sbt-assembly` on a project that contained the `http4s-swagger` routes. I had some additional conflicts that I needed to work through, and thought that I would propose my work here as a part of the docs. Feel free to update this or ignore it.

### CODE REVIEW

File: modules/docs/markdown/01-overview/05-sharing-specs.md [hunk 1]
```diff
@@ -200,14 +200,21 @@ will result in the `"software.amazon.smithy" % "smithy-aws-iam-traits" % "1.14.1
...
-
-One side-effect of this is that if you produce JARs containing artifacts produced by Smithy4s code generation, they'll contain a `smithy4s.tracking.smithy`. This could be a problematic file if you're using `sbt-assembly` because if you depend on multiple JARs that contain this file, you'll need to write a custom `assemblyMergeStrategy`, like so:
+One side-effect of this is that if you produce JARs containing artifacts produced by Smithy4s code generation, they'll contain some common files with other smithy projects, particularly within the `META-INF` folder. You'll need to write a custom `assemblyMergeStrategy`, like so:
 
 ```sbt
 assemblyMergeStrategy := {
-  case "META-INF/smithy/smithy4s.tracking.smithy" =>
-    MergeStrategy.discard
+  case PathList("META-INF", xs @ _*) =>
+        (xs map { _.toLowerCase }) match {
```
Comment (kubukoz) [inline modules/docs/markdown/01-overview/05-sharing-specs.md]: I think these parens are redundant

```suggestion
        xs map { _.toLowerCase } match {
```

also, there's a bit too much nesting in these new lines

Comment (etspaceman) [reply modules/docs/markdown/01-overview/05-sharing-specs.md]: Committed the suggestion. WDYM by too much nesting? Happy to update as needed.

Comment (kubukoz) [reply modules/docs/markdown/01-overview/05-sharing-specs.md]: right, I meant indentation 😅

Comment (kubukoz) [reply modules/docs/markdown/01-overview/05-sharing-specs.md]: image

it's a minor thing really, but it stands out - the lines inside the pattern match body should have 4 spaces of leading whitespace

**Review by kubukoz — APPROVED** [review]

**Comment by kubukoz** [comment]
This may also be useful in a generic "packaging your application" page - it's not specific to sharing specs to downstream projects, but getting fatjars in general.

Maybe that whole section about assembly should go to that new page, and we'd just link to it here - some ideas for a follow-up PR if you feel like making one

**Comment by etspaceman** [comment]
Yeah I could absolutely do that :)

**Pull request merged**